Jenkinsfile

```diff
@@ -16,84 +16,82 @@ pipeline {
             environment {
                 DOCKER_CREDENTIALS_ID = "dockerhub_dopplerdock"
                 DOCKER_IMAGE_NAME = "dopplerdock/doppler-webapp"
             }
-            when {
-                expression {
-                    return isVersionTag(readCurrentTag())
+            stages {
+                stage('Publish pre-release images from pull request') {
+                    when {
+                        changeRequest target: 'master'
+                    }
+                    steps {
+                        sh '''docker build \
+                            -t "local-doppler-webapp:production-commit-${GIT_COMMIT}" \\
+                            --build-arg environment=production \\
+                            --build-arg react_app_router=browser \\
+                            --build-arg public_url="/" \\
+                            --build-arg version=production-${TAG_NAME}+${GIT_COMMIT} \\
+                            -f Dockerfile.swarm \\
+                            .'''
+                    }
                 }
-            }
-            steps {
-                withDockerRegistry(credentialsId: "${DOCKER_CREDENTIALS_ID}", url: "") {
-                    sh 'sh publish-commit-image-to-dockerhub.sh production ${DOCKER_IMAGE_NAME} ${GIT_COMMIT} ${TAG_NAME}'
-                    sh 'sh publish-commit-image-to-dockerhub.sh demo ${DOCKER_IMAGE_NAME} ${GIT_COMMIT} ${TAG_NAME}'
-                    sh 'sh publish-commit-image-to-dockerhub.sh int ${DOCKER_IMAGE_NAME} ${GIT_COMMIT} ${TAG_NAME}'
-                    sh 'sh publish-commit-image-to-dockerhub.sh qa ${DOCKER_IMAGE_NAME} ${GIT_COMMIT} ${TAG_NAME}'
+                stage('Publish pre-release images from master') {
+                    when {
+                        branch 'master'
+                    }
+                    steps {
+                        withDockerRegistry(credentialsId: "${DOCKER_CREDENTIALS_ID}", url: "") {
+                            sh '''docker build \
+                                -t "local-doppler-webapp:qa-commit-${GIT_COMMIT}" \\
+                                --build-arg environment=qa \\
+                                --build-arg react_app_router=browser \\
+                                --build-arg public_url="/" \\
+                                --build-arg version=qa-${TAG_NAME}+${GIT_COMMIT} \\
+                                -f Dockerfile.swarm \\
+                                .'''
+                            sh 'sh publish-commit-image-to-dockerhub.sh qa ${DOCKER_IMAGE_NAME} ${GIT_COMMIT} ${TAG_NAME}'
+                        }
+                    }
+                }
+                stage('Publish pre-release images from INT') {
+                    when {
+                        branch 'INT'
+                    }
+                    steps {
+                        withDockerRegistry(credentialsId: "${DOCKER_CREDENTIALS_ID}", url: "") {
+                            sh '''docker build \
```

Comment:
Antes de mergear, podrías probar de pushear la rama de este PR a INT a ver si la imagen se publica y está disponible correctamente en el entorno de Integración.

Reply:
**Push a INT**
<img width="1162" alt="Captura de Pantalla 2022-09-01 a la(s) 3 19 08 p  m" src="https://user-images.githubusercontent.com/84402180/187995244-ed2b61d1-b647-4d1b-9e5b-46c045438539.png">

En lugar de correr el script de `publish-commit-image-to-dockerhub.sh` que estaba en la webapp, corri el script de `build-n-publish.sh` con la configuración que tiene en el billing-user-api [here](https://github.com/FromDoppler/doppler-billing-user/blob/main/build-n-publish.sh)

Reply:
Mmm, cuidado, creo que eso también publica al CDN

Reply:
Para QA también aplicaste ese cambio? Esto implica hacer un cambio en el repo de Swarm

Reply:
decís que lo mejor sería continuar usando `publish-commit-image-to-dockerhub.sh`?

Reply:
No, no, no me metí tanto como para asegurar eso. Tal vez haya que desarmar esos scripts, por ahí se puede seguir exactamente el mismo esquema que con los microservicios.